pkg/internal/codegen/parse/crd.go

```diff
@@ -470,6 +473,30 @@ func (b *APIs) parseObjectValidation(t *types.Type, found sets.String, comments
 	return props, buff.String()
 }
 
+// parseNativeObjectValidation returns a JSONSchemaProps object and its serialization in
+// Go that describe the validations for the given native object type.
+func (b *APIs) parseNativeObjectValidation(t *types.Type, found sets.String, comments []string, isRoot bool) (v1beta1.JSONSchemaProps, string) {
+	buff := &bytes.Buffer{}
+	props := v1beta1.JSONSchemaProps{
+		Type:        "object",
+		Description: parseDescription(comments),
+	}
+
+	for _, l := range comments {
+		getValidation(l, &props)
+	}
+
+	m, result, required := b.getMembers(t, found)
+	props.Properties = m
+	props.Required = required
+
+	if err := objectTemplate.Execute(buff, objectTemplateArgs{props, result, required, isRoot}); err != nil {
+		log.Fatalf("%v", err)
```

Comment:
Why doesn't the function return `err` and let the caller decide what the policy should be?

Though I later noticed that `typeToJSONSchemaProps()` doesn't return an error. hmm.